utils/docify/docify/docify.py

```diff
@@ -0,0 +1,82 @@
+import os
+import sys
+
+
+def docify(stubfile, class_, method, doc, verbose):
+    if not os.path.exists(stubfile):
+        raise Exception(f'Missing stub file {stubfile}')
+    with open(stubfile) as f:
+        stublines = f.readlines()
+
+    lookfor = f'class {class_}' if class_ else f'def {method}('
+    i = 0
+    in_doc = False
+    while i < len(stublines):
+        line = stublines[i]
+
+        # Skip docstrings
+        l = line.strip()
+        if l == '"""':
+            in_doc = not in_doc
+            i += 1
+            continue
+        if in_doc:
+            i += 1
+            continue
+
+        if line.startswith(lookfor):
+            if class_:
+                i += 1
+                lookfor = f'def {method}('
+                while i < len(stublines):
+                    line = stublines[i].strip()
+
+                    if line == '"""':
+                        in_doc = not in_doc
+                        i += 1
+                        continue
+                    if in_doc:
+                        i += 1
+                        continue
```

Comment:
Wondering if this duplicated code could be eliminated? I was anticipating that after lookfor was changed, we'd go back to the outer loop.

Maybe adding an enum to track what stage we are in (finite state machine) would help do things that are stage-specific like the "method not found in class" exception.

Reply:
I agree a FSM would make it clearer. When I embarked on this I didn't realize the docstrings would break the parser, but they do, because some of the examples in pandas docstrings include 'class'. So the docstring skipping was a late addition and added complexity, but I don't think it merits the effort of rewriting it all as an FSM at this point; maybe in the future. I'd rather spend my time on the pandas stubs themselves.